examples/example_4.py

```diff
@@ -28,38 +28,48 @@ def example_simple_tddft(orca_code, nproc, submit=True, opt_calc_pk=None):
             'input_blocks': {
                 'scf': {
                     'convergence': 'tight',
+                    'moinp': '"aiida_old.gbw"',
                 },
                 'pal': {
                     'nproc': nproc,
                 },
                 'tddft': {
-                    'nroots': 8,
-                    'maxdim': 2,
-                    'triplets': 'true',
-                }
+                    'nroots': 3,
+                    'triplets': 'false',
+                    'tda': 'true',
+                },
             },
-            'input_keywords': ['RKS', 'PBE', 'SV(P)'],
+            'input_keywords': ['RKS', 'BP', 'STO-3G', 'MOREAD'],
             'extra_input_keywords': [],
         }
     )
 
     # Construct process builder
     builder = OrcaCalculation.get_builder()
 
-    builder.structure = load_node(opt_calc_pk).outputs.relaxed_structure
+    # old gbw file
+    opt_calc = load_node(opt_calc_pk)
+    retr_fldr = opt_calc.outputs.retrieved
+    with retr_fldr.open('aiida.gbw', 'rb') as handler:
+        gbw_file = SinglefileData(handler)
```

Comment:
This is a side note, but seeing this, would it maybe make sense to have the parser always attach `aiida.gbw` as a `SinglefileData` output node, instead of storing it in the retrieved folder? Given that the `OrcaCalculation` takes this as an input for restarts? Would simplify user code at least and also make the provenance clearer

Reply:
But you are anyway storing them already as they are part of the `retrieved` node, which is also stored locally. So either this should not be retrieved by default (and a switch in the inputs can let a user tell the plugin to retrieve it when needed) or it might as well be attached as a node, which wouldn't change the load on the database.

Reply:
Okay, so to double check I understand your suggestion, you would attach this file directly as a `SinglefileData`, and removed it from the retrieved folder? I am not sure about the details of AiiDA backed, is SinglefileData essentially just a link to a file storage (so the means of how these files were stored would be identical in the backend?

One thing I am worried about, I am not sure if it is possible now to delete files from the retrieved folder, without AiiDA complaining about provenance? If that is the case, I think the current design is good since users might want to delete these files to save space. If this files were attached dirrectly as output nodes, I don't think one could easily get rid of them, right?

(in any case, this is a discussion for another PR and should not block this one)

Reply:
> I am not sure about the details of AiiDA backed, is SinglefileData essentially just a link to a file storage (so the means of how these files were stored would be identical in the backend?

Essentially it is almost the same. In one case you have a `FolderData` node and in the other a `SinglefileData` node. The former can store any number of files, but the `SinglefileData` just the one. The files itself are essentially stored in the same way. It is just that if a restart calculation requires this file to be passed as a `SinglefileData`, it would make sense to already have the parser perform this operation instead of forcing the user to always have this code snippet to do it (which also duplicates the file in a sense).

> One thing I am worried about, I am not sure if it is possible now to delete files from the retrieved folder, without AiiDA complaining about provenance? If that is the case, I think the current design is good since users might want to delete these files to save space. If this files were attached dirrectly as output nodes, I don't think one could easily get rid of them, right?

The `retrieved` output is a `FolderData` node which is, just like all other nodes, immutable so you cannot delete a file. So this would not be a problem. If it really is a problem that this file is always retrieved and stored (as is currently the case) you could change the code to not retrieve it by default, but have an input to the calculation that will trigger to retrieve it. Historically in `aiida-quantumespresso` this was done through the `settings` input node, which could be used to change the behavior of the calculation. You could think of adding something like `retrieve_gbw`. The `OrcaCalculation` could check this key in the `settings` node, and when set to `True` add the `aiida.gbw` file to the `retrieve_temporary_list`. The `OrcaParser` can then wrap the file from the retrieved temporary folder in a `SinglefileData` node.

Reply:
Would be happy to make a PR for this to show the concept if you and @pzarabadip think it makes sense to have a switch to retrieve (and potentially attach as `SinglefileData`) the file.

Reply:
Thank you for the explanation! Yes, I think having it be optional definitely makes sense.

> Historically in aiida-quantumespresso this was done through the settings input node, which could be used to change the behavior of the calculation. You could think of adding something like retrieve_gbw. The OrcaCalculation could check this key in the settings node, and when set to True add the aiida.gbw file to the retrieve_temporary_list. The OrcaParser can then wrap the file from the retrieved temporary folder in a SinglefileData node.

Yep, that makes sense for sure. The only question would than be what should be the default. Just to make sure, if the file is both in SinglefileData node and FolderData, it won't be duplicated then? (I guess in AiiDA 2.0 which automaticaly deduplicates files that makes sense, not sure for AiiDA 1.x).

> Would be happy to make a PR for this to show the concept if you and @pzarabadip think it makes sense to have a switch to retrieve (and potentially attach as SinglefileData) the file.

Thanks, much appreciated! I think for now I'd like to focus on merging the existing PRs and then we can come back to this. I definitely think it's a good idea. Maybe for now you could open an issue and just copy over the info that you provided above?

Reply:
> Just to make sure, if the file is both in SinglefileData node and FolderData, it won't be duplicated then? (I guess in AiiDA 2.0 which automaticaly deduplicates files that makes sense, not sure for AiiDA 1.x).

No, I would keep it either in the `FolderData` or just as the `SinglefileData`, not in both. In 1.x it would indeed duplicate the file in the repository. In 2.x it would indeed be automatically deduplicated, but that won't be the case for all storage backends (which are pluginnable as of v2.0).

I just read a bit on ORCA and it seems that the .gbw file is the wavefunction. It looks to me that maybe it should not be retrieved by default. We do the same with `aiida-quantumespresso`. The wavefunctions are never retrieved because they are huge. If you want to restart, you simply use the `remote_folder` of the original calculation. I think we should apply this approach for this plugin as well. We can then add the option to retrieve it as a single file in case people want to store a wavefunction locally. This can be useful if the calculation needs to be restartable even in a long time when the scratch on the remote will most likely already have been cleaned.

But I agree, this all not necessary for this PR.

Reply:
I agree with making the retrieve option upon user request. I had in mind (at the time) to use it for conversion to `wfx` file to run `QTAIM` calculation but have not done it after all. Either way, it should be easily doable with `remore_folder` too.